File: MPI_Code/C/sendrecv.c
#include <stdio.h>
#include "mpi.h"
int main()
{
  int myrank, A[4]={0,},B[4]={0,},inext,iprev, i;
  MPI_Init(NULL,NULL);
  MPI_Comm_rank(MPI_COMM_WORLD,&myrank);
  A[myrank]=myrank+1;
  inext=myrank+1,   iprev=myrank-1;
  if(myrank==0) iprev=3;
  if(myrank==3) inext=0;
  if(myrank==0) printf("Before\n");
  for(i=0;i<4;i++)
    if(myrank==i) printf("Myrank: %d, A=%d %d %d %d\n",myrank,A[0],A[1],A[2],A[3]);
  MPI_Sendrecv(&A[myrank],1,MPI_INT,inext,0,&B[myrank],1,MPI_INT,iprev,0,MPI_COMM_WORLD,MPI_STATUS_IGNORE);

  if(myrank==0) printf("After\n");
  for(i=0;i<4;i++)
    if(myrank==i) printf("Myrank: %d, B=%d %d %d %d\n",myrank,B[0],B[1],B[2],B[3]);
  MPI_Finalize();
  return 0;
}

File: example_code/mpi/MPI_Code/Fortran/sendrecv.f90
PROGRAM sendrecv
USE mpi_f08
IMPLICIT NONE
INTEGER::myrank
INTEGER::A(0:3),B(0:3), inext, iprev, i
CALL MPI_Init
CALL MPI_Comm_rank(MPI_COMM_WORLD,myrank)
A=0;   B=0
A(myrank)=myrank+1
inext=myrank+1;   iprev=myrank-1
IF(Myrank==0) iprev=3
IF(Myrank==3) INEXT=0
IF(myrank==0) print*,'BEFORE'
DO i=0,3
  if(myrank==i) print*,'Myrank:',myrank, 'A=',A
END DO
CALL MPI_Sendrecv(A(myrank),1,MPI_INTEGER,INEXT,0,B(myrank),1,MPI_INTEGER,iprev,0,MPI_COMM_WORLD,MPI_STATUS_IGNORE)
IF(myrank==0)THEN
  PRINT*,''
  PRINT*,'AFTER'
END IF
DO I=0,3
  IF(Myrank==i) PRINT*,'Myrank:',myrank,'B=',B
END DO
CALL MPI_Finalize
END PROGRAM sendrecv
